

syntax = "proto3";
package DataJoin.common;

enum StatusCode {
  STATUS_SUCCESS = 0;
  STATUS_UNKNOWN_ERROR = 1;
  STATUS_INVALID_REQUEST = 2;
  STATUS_NO_MORE_DATA = 3;
}

message Status {
  StatusCode code = 1;
  string error_message = 2;
}

enum FLRole {
  Leader = 0;
  Follower = 1;
}

message Data {
    string key = 1;
    bytes value = 2;
    string operator = 3;
}

enum DataSourceState {
  UnKnown = 0;
  Init = 1;
  Processing = 2;
  Ready = 3;
  Finished = 4;
  Failed = 5;
}

message HeaderData {
    string operator = 1;
}

message DataSourceMeta {
  string name = 1;
  uint32 partition_num = 2;
  int64 start_time = 3;
  int64 end_time = 4;
  float negative_sampling_rate = 5;
}

message Packet {
    HeaderData header = 1;
    Data body = 2;
}

message DataSource {
  string data_source_name = 1;
  string data_block_dir = 2;
  string raw_data_dir = 3;
  uint32 partition_id = 4;
  string mode = 5;
  FLRole role = 6;
  DataSourceState state = 7;
}

service ProxyDataService {
    rpc UnaryCall (Packet) returns (Packet);
}


